sdk/identity/azure-identity/src/test/java/LiveManagedIdentityTests.java

```diff
@@ -0,0 +1,192 @@
+        String secret = configuration.get("IDENTITY_CLIENT_SECRET");
+        String tenantId = configuration.get("IDENTITY_TENANT_ID");
+        String resourceGroup = configuration.get("IDENTITY_RESOURCE_GROUP");
+        String subscriptionId = configuration.get("IDENTITY_SUBSCRIPTION_ID");
+        String vmName = configuration.get("IDENTITY_VM_NAME");
+        String storageAcccountName = configuration.get("IDENTITY_STORAGE_NAME_1");
+
+        boolean isWindows = os.contains("Windows");
+
+        String azPath = runCommand(isWindows ? "where" : "which", "az").trim();
+        azPath = isWindows ? extractAzCmdPath(azPath) : azPath;
+
+        runCommand(azPath, "login",  "--service-principal", "-u", spClientId, "-p", secret, "--tenant", tenantId);
+        runCommand(azPath, "account", "set", "--subscription", subscriptionId);
+
+        String vmBlob = String.format("https://%s.blob.core.windows.net/vmcontainer/testfile.jar", storageAcccountName);
+        String script = String.format("curl '%s' -o ./testfile.jar && java -jar ./testfile.jar", vmBlob);
+
+        String output = runCommand(azPath, "vm", "run-command", "invoke", "-n", vmName, "-g", resourceGroup,
+            "--command-id", "RunShellScript", "--scripts", script);
+
+        Assertions.assertTrue(output.contains("Successfully retrieved managed identity tokens"),
+            "Failed to get response from VM");
+    }
+
+    @Test
+    @EnabledIfEnvironmentVariable(named = "AZURE_TEST_MODE", matches = "LIVE")
+    public void callGraphWithClientSecret() {
+        Configuration configuration = Configuration.getGlobalConfiguration().clone();
+
+        String multiTenantId = configuration.get("AZURE_IDENTITY_MULTI_TENANT_TENANT_ID");
+        String multiClientId = configuration.get("AZURE_IDENTITY_MULTI_TENANT_CLIENT_ID");
+        String multiClientSecret = configuration.get("AZURE_IDENTITY_MULTI_TENANT_CLIENT_SECRET");
+
+        ClientSecretCredential credential = new ClientSecretCredentialBuilder()
+            .tenantId(multiTenantId)
+            .clientId(multiClientId)
+            .clientSecret(multiClientSecret)
+            .build();
+
+
+        AccessToken accessToken = credential
+            .getTokenSync(new TokenRequestContext().addScopes("https://graph.microsoft.com/.default"));
+
+        Assertions.assertTrue(accessToken != null, "Failed to get access token");
+    }
+
+    private String runCommand(String... args) {
+        try {
+            StringBuilder command = new StringBuilder();
+            for (String arg : args) {
+                command.append(arg).append(" ");
+            }
+            System.out.println("Running command: " + command);
+            ProcessBuilder processBuilder = new ProcessBuilder(args);
+            processBuilder.redirectErrorStream(true);
+            Process process = processBuilder.start();
+            // Set timeout
+            boolean finished = process.waitFor(300, TimeUnit.SECONDS);
```

Comment:
Instead of a timeout here should we just apply a `@Timeout` to the test class?

Reply:
Added it